pkg/build/builder/sti_test.go

```diff
@@ -145,3 +153,65 @@ func TestCopyToVolumeList(t *testing.T) {
 		t.Errorf("Expected artifacts mapping to match %#v, got %#v instead!", volumeList, newVolumeList)
 	}
 }
+
+func TestBuildEnvVars(t *testing.T) {
+	expectedEnvList := s2iapi.EnvironmentList{
+		s2iapi.EnvironmentSpec{
+			Name:  "HTTP_PROXY",
+			Value: "http://test/insecure:8080",
+		},
+		s2iapi.EnvironmentSpec{
+			Name:  "HTTPS_PROXY",
+			Value: "https://test/secure:8443",
+		},
+		s2iapi.EnvironmentSpec{
+			Name:  "OPENSHIFT_BUILD_NAME",
+			Value: "openshift-test-1-build",
+		},
+		s2iapi.EnvironmentSpec{
+			Name:  "OPENSHIFT_BUILD_NAMESPACE",
+			Value: "openshift-demo",
+		},
+		s2iapi.EnvironmentSpec{
+			Name:  "OPENSHIFT_BUILD_SOURCE",
+			Value: "http://localhost/123",
+		},
+	}
+	mockBuild := makeBuild()
+	mockBuild.Name = "openshift-test-1-build"
+	mockBuild.Namespace = "openshift-demo"
+	resultedEnvList := buildEnvVars(mockBuild)
+	if !reflect.DeepEqual(expectedEnvList, resultedEnvList) {
+		t.Errorf("Expected EnvironmentList to match: %#v, got %#v", expectedEnvList, resultedEnvList)
```

Comment:
@PI-Victor I suspect that the problem here is because of order of elements in the array.